app/lib/three_scale/redis_config.rb

```diff
@@ -42,15 +40,24 @@ def respond_to_missing?(method_sym, *args)
 
     DEFAULT_SENTINEL_PORT = 26379
 
-    def parse_sentinels(sentinels)
+    def parse_sentinels(config)
+      sentinels = config.delete(:sentinels).presence
       return unless sentinels
-      sentinels.to_s.split(',').map do |sentinel_url|
+
+      sentinel_user = nil
+      sentinel_password = nil
+      config[:sentinels] = sentinels.to_s.split(',').map do |sentinel_url|
         uri = URI.parse((sentinel_url =~ %r{^(redis(s)?|unix):\/\/.+} ? '' : 'redis://') + sentinel_url)
-        parsed_sentinel = { host: uri.host, port: (uri.port || DEFAULT_SENTINEL_PORT) }
-        password = uri.password
-        parsed_sentinel[:password] = password if password
-        parsed_sentinel
+        sentinel_user ||= uri.user
+        sentinel_password ||= uri.password
+        { host: uri.host, port: (uri.port || DEFAULT_SENTINEL_PORT) }
```

Comment:
Probably it would be a really bad configuration to have sentinels use different credentials... but I feel like URI should override credentials otherwise set. Or was this the sole issue that this PR is fixing?

Reply:
> Probably it would be a really bad configuration to have sentinels use different credentials

It must be a bad idea because it's forbidden by Redis itself

> but I feel like URI should override credentials otherwise set

I'm not sure what happens when the URI contradicts whatever you set in other ENV variables, but reading the code I would say the URI takes precedence because [`parse_sentinels`](https://github.com/jlledom/porta/blob/THREESCALE-11205-sentinels-url-credentials/app/lib/three_scale/redis_config.rb#L11) is the last method to be called when reading the Redis config.

> Or was this the sole issue that this PR is fixing?

The purpose of this PR is to fix how the sentinel credentials are extracted from the URL, because the old code ended up creating an options hash which wasn't compatible with `redis-rb` after upgrading to version 5.

Reply:
Got it,  sounds reasonable! Is this behavior tested - that URL overrides the other config? Just the point is to have this behavior fixed for the future.

Reply:
I made some tests. For the sentinels credentials, the URL overrides the env variables, for some other parameters the env variables override the URL, so there's no consistence. That's because I didn't care of that when I implemented TLS for Redis. It would be good to implement a consistent precedence on redis params, i.e. URL always over env varaibles, but considering the hurry, I would leave that for another PR.

Reply:
The issue with such approach is that users may try to use in a different way and then be badly surprised on upgrade. I think it is better to get it as we want before the testing cycle even if it takes a day more.

Reply:
Yes, I created this issue: https://issues.redhat.com/browse/THREESCALE-11216 and it would be good to do it soon. If we don't do it before the release, I don't think it'll be much of a problem anyway because the clients aren't supposed to use the env variables, those are only for the operator, and we can work with the operator engineers to find a way to make it compatible between versions.